### PR TITLE
Refactor blocked segments

### PR DESCRIPTION
**Pull request by Gymnasiast**
This saves a few thousand lines mainly by using a constant for the blocked segments of straight flat and diagonal flat rides. It also includes some consolidation of support calls, further reducing line count.

Using constants for blocked segments is another small step in genericising the paint code.

### CODE REVIEW

File: src/openrct2/paint/track/Segment.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const uint16_t kStraightFlat = EnumsToFlags(PaintSegment::centre, PaintSegment::topRightSide, PaintSegment::bottomLeftSide);
+    const uint16_t kDiagStraightFlat[] = {
+        EnumsToFlags(
+            PaintSegment::centre, PaintSegment::topRightSide, PaintSegment::bottomRightSide, PaintSegment::rightCorner),
+        EnumsToFlags(PaintSegment::centre, PaintSegment::topRightSide, PaintSegment::topLeftSide, PaintSegment::topCorner),
+        EnumsToFlags(
+            PaintSegment::bottomLeftSide, PaintSegment::centre, PaintSegment::bottomCorner, PaintSegment::bottomRightSide),
+        EnumsToFlags(PaintSegment::bottomLeftSide, PaintSegment::centre, PaintSegment::leftCorner, PaintSegment::topLeftSide),
+    };
+} // namespace BlockedSegments
```
Comment (ZehMatt) [inline src/openrct2/paint/track/Segment.cpp:23]: I think its best to have this in the header and declare it constexpr, this is not a lot of data and I don't think having getter functions for those makes much sense.

**Review by ZehMatt — CHANGES_REQUESTED** [review]
Left a small remark, I think its overall fine but ideally there is some solid testing.

File: src/openrct2/ride/thrill/3dCinema.cpp [hunk 2]
```diff
@@ -11,6 +11,8 @@
 #include "../../interface/Viewport.h"
 #include "../../paint/Paint.h"
 #include "../../paint/support/WoodenSupports.h"
+#include "../../paint/tile_element/Segment.h"
+#include "../../paint/track/Segment.h"
```
Comment (karst) [inline src/openrct2/ride/thrill/3dCinema.cpp:15]: In flat rides and such these entries aren't actually used. Are they just here for consistency sake?

**Review by karst — APPROVED** [review]
My apologies for letting you wait for longer than I promised.
Here is the results of my playtesting:
After extensive park checking and checking each ride type on the track zoo my results are that: 
- No oddities found

File: src/openrct2/ride/thrill/3dCinema.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/thrill/3dCinema.cpp:15]: Good catch. They do use `kSegmentsAll`, though, but I can still drop `track/Segment.h`.

**Comment by spacek531** [comment]
Issue with Inverted Coaster DiagFlat, support is not blocked by track piece:
image

Same issue with Compact Inverted Coaster DiagFlat:
image

Same issue with Mini Suspended Coaster DiagFlat:
image

Same issue with Suspended Monorail DiagFlat: 
image

Same issue with Suspended Swinging Coaster DiagFlat:
image

Same issue with Lay-Down Coaster DiagFlat:
image

Rides tested:
- Mini Coaster
- Inverted Coaster
- LIM Launched Coaster
- Compact Inverted Coaster
- Mini Suspended Coaster
- Suspended Monorail
- Twister Coaster
- Vertical Drop Coaster
- Suspended Swinging Coaster

Test procedure:

- load park file [TrackZoo 2.zip](https://github.com/OpenRCT2/OpenRCT2/files/15185126/TrackZoo.2.zip)
- create giant screenshot using develop build
- create giant screenshot using feature build
- layer screenshots using paint.NET
- set upper layer to XOR blending
- scan for non-black pixels

Tested individually by building:
- Multi-Dimension Coaster
- Flying Coaster
- Lay-Down Coaster

**Review by spacek531 — APPROVED** [review]
I have re-tested the following coasters:
- mini coaster
- compact inverted coaster
- inverted coaster
- looping coaster
- mini suspended coaster
- suspended monorail
- suspended swinging coaster
- vertical drop coaster
- lay-down coaster
- flying coaster
- multi-dimension coaster

No changes were observed between giant screenshots in OpenRCT2, v0.4.10 (e55d761) provided by GitHub and feature branch.

Great job!

**Pull request merged**